### PR TITLE
New endpoint for Delta: createStagingTable

### PR DESCRIPTION
**Pull request by yili-db**
## 🥞 Stacked PR
Use this [link](https://github.com/unitycatalog/unitycatalog/pull/1511/files) to review incremental changes.
- [**stack/staging_tables**](https://github.com/unitycatalog/unitycatalog/pull/1511) [[Files changed](https://github.com/unitycatalog/unitycatalog/pull/1511/files)]
  - [stack/create_table](https://github.com/unitycatalog/unitycatalog/pull/1516) [[Files changed](https://github.com/unitycatalog/unitycatalog/pull/1516/files/4530c551f0f7048a0bb96aeb590a51f4de096ffe..43c349a739ac803aeebb736ea7f500222187749b)]
    - stack/create_table_uniform
    - stack/drc_update_table

---------
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**
New endpoint for Delta: createStagingTable

Adds POST /v1/catalogs/{catalog}/schemas/{schema}/staging-tables. The response carries the UC-allocated tableId, a managed storage location, READ_WRITE credentials, and the UC catalog-managed Delta contract (required + suggested protocol features and properties) that the external writer must honor in the initial commit.

UcManagedDeltaContract is the single source of truth for this contract; DeltaStagingTableMapper reads from it. Future create/update/commit endpoints will validate against the same constants so producer and consumer cannot drift. DeltaConsts centralizes Delta protocol identifiers (TableFeature enum with TableFeatureKind classification, TableProperties keys); utils/TableProperties.java is deleted, its constants fold into DeltaConsts.TableProperties (wire strings unchanged).

AuthorizeExpressions.CREATE_STAGING_TABLE is shared between UC REST StagingTableService and the new DRC endpoint -- compile-time guarantee both surfaces enforce identical authz. SdkStagingTableAccessControlTest covers both arms of the OR plus a deny case; a roundtrip test pins null map values surviving NON_NULL Jackson.

### CODE REVIEW

File: server/src/main/java/io/unitycatalog/server/service/delta/DeltaStagingTableMapper.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+          "delta.checkpoint.writeStatsAsStruct", "true",
+          "delta.enableDeletionVectors", "true",
+          "delta.enableInCommitTimestamps", "true");
+
+  // Required properties that the engine must compute at commit time. Null value here means
+  // "must be present in the initial commit, UC does not constrain the value" per the
+  // StagingTableResponse spec. Both are engine-managed and tied to the inCommitTimestamp feature.
+  private static final List<String> ENGINE_GENERATED_REQUIRED_PROPERTY_KEYS =
+      List.of(
+          "delta.inCommitTimestampEnablementVersion", "delta.inCommitTimestampEnablementTimestamp");
```
Comment (yili-db) [inline server/src/main/java/io/unitycatalog/server/service/delta/DeltaStagingTableMapper.java]: raw strings in constant.

File: server/src/main/java/io/unitycatalog/server/service/delta/DeltaTableFeatures.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Delta protocol table-feature identifiers referenced by the Delta REST Catalog surface. These are
+ * the spec-defined feature names that appear in {@code protocol.reader-features} and {@code
+ * protocol.writer-features}.
+ *
+ * <p>Centralising them prevents typos at the points where the server makes feature-specific
+ * decisions (e.g. "MANAGED tables must declare catalogManaged") from silently mismatching the
+ * features the client actually wrote into the Delta log.
+ */
+public final class DeltaTableFeatures {
```
Comment (tdas) [inline server/src/main/java/io/unitycatalog/server/service/delta/DeltaTableFeatures.java]: DeltaConsts.Features.xxx
DeltaConsts.Properties.xxx

**Review by tdas — APPROVED** [review]
just one comment.

**Review by raveeram-db — APPROVED** [review]

File: server/src/main/java/io/unitycatalog/server/service/delta/DeltaConsts.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+   *
+   * <ul>
+   *   <li>{@link #READER_WRITER}: feature must appear in BOTH {@code protocol.reader-features} and
+   *       {@code protocol.writer-features}.
+   *   <li>{@link #WRITER_ONLY}: feature appears only in {@code protocol.writer-features}.
+   * </ul>
+   *
+   * Delta has no reader-only features.
+   */
+  public enum TableFeatureKind {
```
Comment (openinx) [inline server/src/main/java/io/unitycatalog/server/service/delta/DeltaConsts.java:24]: Sorry, I did not get how will the `TableFeatureKind` affect the behavior ?

File: server/src/main/java/io/unitycatalog/server/service/delta/DeltaRestCatalogService.java [hunk 4]
```diff
@@ -111,6 +122,44 @@ public LoadTableResponse loadTable(
     return tableRepository.loadTableForDelta(catalog, schema, table);
   }
 
+  // ==================== Staging Table API ====================
+
+  /**
+   * Create a staging table for a soon-to-be-created UC catalog-managed Delta table. UC allocates a
+   * UUID and a managed storage location; the client then writes the initial Delta commit to that
+   * location and calls {@code POST /tables} to finalize the table.
```
Comment (openinx) [inline server/src/main/java/io/unitycatalog/server/service/delta/DeltaRestCatalogService.java:130]: > calls {@code POST /tables} to finalize the table.

Is the POST call exactly the createTable API ?  will it be more clear to explicitly say the `createTable` API ?

File: server/src/main/java/io/unitycatalog/server/service/delta/UcManagedDeltaContract.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+  public static final List<String> SUGGESTED_WRITER_FEATURES = writerFeaturesOf(SUGGESTED_FEATURES);
+
+  /**
+   * Required properties with fixed values. {@code io.unitycatalog.tableId} is filled in per-request
+   * from the UC-allocated table UUID and is checked separately by the consumer.
+   */
+  public static final Map<String, String> REQUIRED_FIXED_PROPERTIES =
+      Map.of(
+          TableProperties.CHECKPOINT_POLICY, "v2",
+          TableProperties.CHECKPOINT_WRITE_STATS_AS_JSON, "false",
```
Comment (openinx) [inline server/src/main/java/io/unitycatalog/server/service/delta/UcManagedDeltaContract.java]: According to my recent performance tests, I think the oss delta actually writte the stats as JSON and STRUCT as the same time.  you can search the delta codebase to confirm.  
with this context,  do we need to set this to `true` by default ?

File: server/src/main/java/io/unitycatalog/server/service/StagingTableService.java [hunk 6]
```diff
@@ -36,13 +35,14 @@ public StagingTableService(
     this.schemaRepository = repositories.getSchemaRepository();
   }
 
+  static final String disableGoogleJavaFormat =
```
Comment (openinx) [inline server/src/main/java/io/unitycatalog/server/service/StagingTableService.java:38]: why add this ?

**Review by openinx — APPROVED** [review]
Most are minor comments.  Pls particularly check the `writeStatsAsJson` default value.  I think current behaivor is: write stats as json and struct at the same time. 

Others look good to me.

File: server/src/main/java/io/unitycatalog/server/service/delta/UcManagedDeltaContract.java (hunk 5 above)
Comment (yili-db) [reply server/src/main/java/io/unitycatalog/server/service/delta/UcManagedDeltaContract.java]: Removed. Both are true by default in Delta Spark.

File: server/src/main/java/io/unitycatalog/server/service/StagingTableService.java (hunk 6 above)
Comment (yili-db) [reply server/src/main/java/io/unitycatalog/server/service/StagingTableService.java:38]: This is a hack to prevent Google Java Formatter to automatically format this file to a state that checkstyle would fail.

File: server/src/main/java/io/unitycatalog/server/service/delta/DeltaConsts.java (hunk 3 above)
Comment (yili-db) [reply server/src/main/java/io/unitycatalog/server/service/delta/DeltaConsts.java:24]: Some features are both reader and writer features, while others are writer only.
This affects how they appear in which of the feature list in `required_protocol`.
